src/crab/extrapolator.cpp

```
@@ -0,0 +1,43 @@
+bool Extrapolator::ascend(EbpfDomain new_pre) {
+    if (new_pre <= invariant_) {
+        invariant_ = std::move(new_pre);
+        return true;
+    }
+    ++ascending_iteration_;
+    if (ascending_iteration_ < widening_delay_) {
+        invariant_ = invariant_ | new_pre;
+    } else {
+        invariant_ = invariant_.widen(new_pre);
+        if (ascending_iteration_ == widening_delay_) {
+            invariant_ = invariant_ & constant_limits_;
```

Cppcheck (2.20.0)

[style] The function 'to_unsigned' is never used.

(unusedFunction)